flake.nix

```diff
@@ -62,8 +71,85 @@
               fourmolu
               haskell-language-server
               hlint
-            ]);
+            ])++ [ cardanoPkgs.cardano-node cardanoPkgs.cardano-cli ] ++ [pkgs.postgresql];
         });
+      testnet = perSystem (system:
+        let pkgs =nixpkgsFor system;
+        in
+          pkgs.runCommand "setPrivateNetwork" {buildInputs=(with pkgs; [git]);} (builtins.readFile ./setPrivateNetwork.sh)
+        );
+      runTestnet = self.testnet.x86_64-linux.testnet;
+
+      testnet2 = perSystem (system: 
+        let
+          pkgs = nixpkgsFor system;
+          cardanoPkgs = cardanoPkgsFor system;
+          privateNetwork.url = "https://github.com/woofpool/cardano-private-testnet-setup";
+          localPath = "test-env/ogmios-datum-cache-private-network/cardano-private-testnet-setup";
+        in 
+          pkgs.writeShellScript "runTestnet" ''
+          echo "testnet2 begin"          
+          mkdir -p ${localPath}
+          
+          LOCALREPO_VC_DIR=$LOCALREPO/.git
+          
+          if [ ! -d ${localPath}/.git ]
+          then
+              ${pkgs.git} clone ${privateNetwork.url} ${localPath}
```

Comment:
`git clone` is not going to work because there's no network connection. You'd have to make this a fixed-output derivation or set the `__impure` attribute (and enable another experimental feature), but that's not worth it. This should be rewritten to avoid calling `git` directly

Switching to a flake input for `cardano-private-testnet` as I suggest below would allow you to use the files from there. You could do something like

```
cp -r ${inputs.cardano-private-testnet-setup} .
```

You can also copy the scripts from the ODC repo's `test-env` directory:

```
cp -r ${self}/test-env/ogmios-datum-cache-private-network/postgres/* .
```

or something

Reply:
I got a lot of problems with this, It seems `flakes` makes all it can to avoid providing access to the local repo. 

This also made me drop the idea of copying `ogmios-datum-cache-private-network` using nix, instead I would run it in a regular bash script  and put an entry in `Makefile`